packages/suite/src/middlewares/wallet/coinmarketMiddleware.ts

```diff
@@ -146,6 +148,15 @@ export const coinmarketMiddleware =
                     ),
                 );
             }
+
+            if (
+                newState.wallet.coinmarket.prefilledFromCryptoId &&
+                !isSell &&
+                !isExchange &&
+                !isBuy
+            ) {
```

Comment:
It will not clear the property when changing switching between buy and sell section. Please, consider this case. 

```suggestion
            if (
                newState.wallet.coinmarket.prefilledFromCryptoId &&
                (!isSell || !isExchange || !isBuy)
            ) {
```

Reply:
The suggestion did not work for me well. Even moving the logic before action propagation did not work. I believe I solved it in [498ad4b](https://github.com/trezor/trezor-suite/pull/15128/commits/498ad4bf4461a18123a258db24f40c23879738f1)

Reply:
Thank you @LukasRada for the update. Fine by me now!

Reply:
Sorry for bad suggestion.

Reply:
No worries. Thanks for the review :)